internal/provider/resource_role.go

```diff
@@ -164,33 +164,36 @@ func resourceRoleCreate(ctx context.Context, d *schema.ResourceData, meta interf
 	}
 	raw := tcr.GetResponse().Map
 
+	var diags diag.Diagnostics
 	if principalIds != nil {
 		tspr, err := rc.SetPrincipals(ctx, tcr.Item.Id, 0, principalIds, roles.WithAutomaticVersioning(true))
 		if err != nil {
-			return diag.Errorf("error setting principal IDs on role: %v", err)
-		}
-		if tspr == nil {
-			return diag.Errorf("nil role after setting principal IDs")
+			diags = append(diags, diag.Diagnostic{Severity: diag.Error, Summary: "error setting principals", Detail: err.Error()})
+		} else {
+			if tspr == nil {
+				return diag.Errorf("nil role after setting principal IDs")
+			}
+			raw = tspr.GetResponse().Map
 		}
-		raw = tspr.GetResponse().Map
 	}
 
 	if grantStrings != nil {
 		tsgr, err := rc.SetGrants(ctx, tcr.Item.Id, 0, grantStrings, roles.WithAutomaticVersioning(true))
 		if err != nil {
-			return diag.Errorf("error setting grant strings on role: %v", err)
-		}
-		if tsgr == nil {
-			return diag.Errorf("nil role after setting grant strings")
+			diags = append(diags, diag.Diagnostic{Severity: diag.Error, Summary: "error setting grants", Detail: err.Error()})
+		} else {
+			if tsgr == nil {
+				return diag.Errorf("nil role after setting grant strings")
```

Comment:
While this shouldn't happen, we _do_ already have a created role.  Maybe this should just append to diag and continue like the err != nil case?

Here and for principalIds above.

Reply:
Good catch!